src/lib/features/feature-toggle/configuration-revision-service.ts

```diff
@@ -69,8 +93,12 @@ export default class ConfigurationRevisionService extends EventEmitter {
         );
         if (this.revisionId !== revisionId) {
             this.logger.debug(
-                'Updating feature configuration with new revision Id',
-                revisionId,
+                `Updating feature configuration with new revision Id ${revisionId} and all envs: ${Object.keys(this.maxRevisionId).join(', ')}`,
+            );
+            await Promise.allSettled(
+                Object.keys(this.maxRevisionId).map((environment) =>
+                    this.updateMaxEnvironmentRevisionId(environment),
```

Comment:
Again, this is a Map and should be handled as such. I don't think this is doing what we're expecting 🤔 

<img width="201" height="92" alt="image" src="https://github.com/user-attachments/assets/1cdfeaf2-e144-4446-825b-714ae8d6ae6b" />


```
await Promise.allSettled(
    [...this.maxRevisionId.keys()].map((environment) =>
        this.updateMaxEnvironmentRevisionId(environment),
    ),
);
```